### PR TITLE
feat(apigateway): integration timeout

### LINKED ISSUE #14123: (apigateway): support integration timeout
API gateway lets you specify a timeout for integrations. This is available in CloudFormation (see [here](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-apitgateway-method-integration.html#cfn-apigateway-method-integration-timeoutinmillis)). However, this isn't available when using RestApi construct.

### Reproduction Steps

Create any `RestApi` with an integration, can't specify the timeout directly.

### What did you expect to happen?

To be able to specify a custom timeout.

### What actually happened?

Default timeout used.

Can use overrides to achieve (see [here](https://play-with-cdk.com/?s=8a4ae926b2acd896488533f2c96d539f)), but should be directly available:

```typescript
const api = new apig.RestApi(this, "MyRestApi");
const httpIntegration = new apig.HttpIntegration("https://example.com");
const proxy = api.root.addProxy({
   defaultIntegration: httpIntegration,
   anyMethod: true,
});
// Override the timeout on integration since CDK doesn't expose property.
const anyMethod = proxy!.anyMethod;
const cfnAnyMethod = (anyMethod!.node!.defaultChild as apig.CfnMethod);
cfnAnyMethod.addPropertyOverride("Integration.TimeoutInMillis", cdk.Duration.seconds(10).toMilliseconds())
```

### Environment

  - **CDK CLI Version  :**
  - **Framework Version:**
  - **Node.js Version:** 
  - **OS               :**
  - **Language (Version):** 

### Other

--- 

This is :bug: Bug Report

### PR DESCRIPTION
**Pull request by Samrose-Ahmed**
Add an option to set a custom timeout when adding a RestApi integration.

fixes #14123

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

**Comment by Samrose-Ahmed** [comment]
It seems to be complaining about a readme update. Should I change this to a fix, or should I add something to the readme?

File: packages/@aws-cdk/aws-apigateway/lib/integration.ts [hunk 1]
```diff
@@ -80,6 +80,13 @@ export interface IntegrationOptions {
    */
   readonly requestTemplates?: { [contentType: string]: string };
 
+  /**
+   * The maximum amount of time an integration will run before it returns without a response.
+   * Must be between 50 and 29,000 milliseconds.
+   * @default 29,000 milliseconds or 29 seconds.
```
Comment (nija-at) [inline packages/@aws-cdk/aws-apigateway/lib/integration.ts:86]:
```suggestion
   * @default Duration.seconds(29)
```

File: packages/@aws-cdk/aws-apigateway/lib/integration.ts [hunk 2]
```diff
@@ -193,6 +200,10 @@ export class Integration {
     if (options.connectionType === ConnectionType.INTERNET && options.vpcLink !== undefined) {
       throw new Error('cannot set \'vpcLink\' where \'connectionType\' is INTERNET');
     }
+
+    if (options.timeout && (options.timeout.toMilliseconds() < 50 || options.timeout.toMilliseconds() > 29000)) {
+      throw new Error('Integration timeout must be between 50 and 29000 milliseconds.');
```
Comment (nija-at) [inline packages/@aws-cdk/aws-apigateway/lib/integration.ts:205]:
```suggestion
      throw new Error('Integration timeout must be between 50 milliseconds and 29 seconds.');
```

File: packages/@aws-cdk/aws-apigateway/lib/integration.ts [hunk 3]
```diff
@@ -80,6 +80,13 @@ export interface IntegrationOptions {
    */
   readonly requestTemplates?: { [contentType: string]: string };
 
+  /**
+   * The maximum amount of time an integration will run before it returns without a response.
+   * Must be between 50 and 29,000 milliseconds.
```
Comment (nija-at) [inline packages/@aws-cdk/aws-apigateway/lib/integration.ts:85]:
```suggestion
   * Must be between 50 milliseconds and 29 seconds.
```

File: packages/@aws-cdk/aws-apigateway/lib/integration.ts [hunk 4]
```diff
@@ -193,6 +200,10 @@ export class Integration {
     if (options.connectionType === ConnectionType.INTERNET && options.vpcLink !== undefined) {
       throw new Error('cannot set \'vpcLink\' where \'connectionType\' is INTERNET');
     }
+
+    if (options.timeout && (options.timeout.toMilliseconds() < 50 || options.timeout.toMilliseconds() > 29000)) {
```
Comment (nija-at) [inline packages/@aws-cdk/aws-apigateway/lib/integration.ts:204]: You also need to sure that timeout is not a token before applying this validation

```suggestion
    if (options.timeout && !options.timeout.isUnresolved() && (options.timeout.toMilliseconds() < 50 || options.timeout.toMilliseconds() > 29000)) {
```

**Review by nija-at — CHANGES_REQUESTED** [review]
Hi @Samrose-Ahmed -

Thanks for submitting this PR! The code looks good. A few minor comments below.

> It seems to be complaining about a readme update. Should I change this to a fix, or should I add something to the readme?

Usually we expect feature requests to have a corresponding entry in the readme, but since this is trivial, I've added a label that will prevent the linter from complaining.

**Review by nija-at — APPROVED** [review]

**Pull request merged**